### PR TITLE
Fix release expansion issue

### PR DESCRIPTION
**Pull request by zeripath**
Use native makefile foreach expansion instead of bash specific brace expansion

Fix #14578

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by silverwind** [comment]
Better alternative for the `eval` above:

```sh
$(eval ESBUILD_VERSION := $(shell node -p "require('./package-lock.json').dependencies.esbuild.version"))
```

**Comment by silverwind** [comment]
Got below to work. apparenly the `.npmrc` changes were not committed and `make .npm-cache` leaves the files dirty currently:

```diff
diff --git a/.npmrc b/.npmrc
index c5ac9901c..25ed63413 100644
--- a/.npmrc
+++ b/.npmrc
@@ -1,4 +1,5 @@
 audit=false
 fund=false
 package-lock=true
 save-exact=true
+cache=.npm-cache
diff --git a/Makefile b/Makefile
index f2114b643..c99483dec 100644
--- a/Makefile
+++ b/Makefile
@@ -670,9 +670,9 @@ node_modules: package-lock.json
 npm-cache: .npm-cache $(FOMANTIC_WORK_DIR)/node_modules/fomantic-ui

 .npm-cache: package-lock.json
 	rm -rf .npm-cache
-	$(eval ESBUILD_VERSION := `node -p "require('./package-lock.json').dependencies.esbuild.version"`)
+	$(eval ESBUILD_VERSION := $(shell node -p "require('./package-lock.json').dependencies.esbuild.version"))
 	npm config --userconfig=.npmrc set cache=.npm-cache
 	rm -rf node_modules && npm install --no-save
 	npm config --userconfig=$(FOMANTIC_WORK_DIR)/.npmrc set cache=../../.npm-cache
 	echo $(foreach build, darwin-64 $(foreach arch,arm arm64 32 64,linux-${arch}) $(foreach arch,32 64,windows-${arch}), esbuild-${build}@$(ESBUILD_VERSION)) | tr " " "\n" | xargs -n 1 -P 4 npm cache add
diff --git a/web_src/fomantic/.npmrc b/web_src/fomantic/.npmrc
index cc243d01b..5f8054bbb 100644
--- a/web_src/fomantic/.npmrc
+++ b/web_src/fomantic/.npmrc
@@ -1,2 +1,3 @@
 optional=false
 package-lock=false
+cache=../../.npm-cache
```

**Comment by zeripath** [comment]
lol I was just discovering this myself

File: Makefile [hunk 1]
```diff
@@ -669,13 +669,14 @@ node_modules: package-lock.json
 .PHONY: npm-cache
 npm-cache: .npm-cache $(FOMANTIC_WORK_DIR)/node_modules/fomantic-ui
 
+
 .npm-cache: package-lock.json
 	rm -rf .npm-cache
-	$(eval ESBUILD_VERSION := `node -p "require('./package-lock.json').dependencies.esbuild.version"`)
+	$(eval ESBUILD_VERSION := $(shell node -p "require('./package-lock.json').dependencies.esbuild.version" ))
```
Comment (silverwind) [inline Makefile]:
```suggestion
	$(eval ESBUILD_VERSION := $(shell node -p "require('./package-lock.json').dependencies.esbuild.version"))
```

File: Makefile [hunk 2]
```diff
@@ -669,13 +669,14 @@ node_modules: package-lock.json
 .PHONY: npm-cache
 npm-cache: .npm-cache $(FOMANTIC_WORK_DIR)/node_modules/fomantic-ui
 
+
```
Comment (silverwind) [inline Makefile]:
```suggestion
```

**Review by silverwind — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by CL-Jeremy** [comment]
Sorry, those `cache=` lines were not meant to be committed per my design. After all, `.npm-cache` is not included when doing `git clone`, so the cache settings should only take effect when `make npm-cache` is executed manually, which is what happens during the release of tarball.

Could you please revert this last commit @zeripath ? Thanks in advance.

**Comment by silverwind** [comment]
Can we somehow make those not write into the `.npmrc` files? Make targets should not leave a unclean git worktree.

**Comment by zeripath** [comment]
@CL-Jeremy just make another pr removing the .npm-cache files

**Comment by CL-Jeremy** [comment]
> Can we somehow make those npm commands not write into the `.npmrc` files? Make targets should not leave a unclean git worktree.

It's possible to make .npm-cache auto-detected for Makefile (i. e. run `npm` with corresponding arguments if cache exists), but I'd like to have plain `npm` also working to give the user as much freedom as possible.

Is it feasible to make a pre-commit hook to prevent commit when `.npm-cache` exists? That'll be perfect.

**Comment by silverwind** [comment]
> Is it feasible to make a pre-commit hook to prevent commit when .npm-cache exists?

Out of question for me at least, I hate all kinds of pre-commit hooks. I think we should just keep it simple and go with https://github.com/go-gitea/gitea/pull/14578#issuecomment-812508403.

**Pull request merged**